lib/omise/OmiseCapabilities.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+class OmiseCapabilities extends OmiseApiResource
+{
+    const ENDPOINT = 'capability';
+
+    /**
+     * @var array  of the filterable keys.
+     */
+    const FILTERS = array(
+        'backend' => array('currency', 'type', 'chargeAmount')
+    );
+
+    protected function __construct($publickey = null, $secretkey = null)
+    {
+        parent::__construct($publickey, $secretkey);
+        $this->setupFilterShortcuts();
+    }
+
+    /**
+     * Retrieves capabilities.
+     *
+     * @param  string $publickey
+     * @param  string $secretkey
+     *
+     * @return OmiseCapabilities
+     */
+    public static function retrieve($publickey = null, $secretkey = null)
+    {
+        return parent::g_retrieve(get_class(), self::getUrl(), $publickey, $secretkey);
+    }
+
+    /**
+     * (non-PHPdoc)
+     *
+     * @see OmiseApiResource::g_reload()
+     */
+    public function reload()
+    {
+        parent::g_reload(self::getUrl());
+    }
+
+    /**
+     * Retrieves array of payment backends. Optionally pass in as many filter functions as you want
+     * (muliple arguments, or a single array)
+     *
+     * @param [func1,fun2,...] OR func1, func2,...
+     *
+     * @return array
+     */
+    public function getBackends()
+    {
+        $backends = array_map(
+            function ($backend) {
+                $id = array_keys($backend)[0];
+                $backend[$id]['_id'] = $id;
+                return (object)reset($backend);
+            },
+            $this['payment_backends']
+        );
+
+        // check for filters
+        if ($filters = func_get_args()) {
+            $filter   = self::combineFilters(self::argsToVariadic($filters));
+            $backends = array_filter($backends, $filter);
+            sort($backends);
+        }
+
+        return $backends;
+    }
+
+    /**
+     * Makes a filter function to check supported currency for backend.
+     *
+     * @param  string $currency
+     *
+     * @return function
+     */
+    public function makeBackendFilterCurrency($currency)
+    {
+        return function ($backend) use ($currency) {
```

Comment:
Again, why split over many lines? It's like code diarrhoea

Reply:
All of code diarrhoea stuff you commented were changed for the sake of Code Sniffer to make it comply with the code standard.